src/bitcoin_client.rs

```diff
@@ -182,13 +187,51 @@ impl BitcoinClientApi for BitcoinClient {
         tx_id: &Txid,
     ) -> Result<GetRawTransactionResult, BitcoinClientError> {
         let tx = self.client.get_raw_transaction_info(tx_id, None)?;
-        debug!("get_raw_transaction_info({}) -> found: {}", tx_id, tx.txid == *tx_id);
+        debug!(
+            "get_raw_transaction_info({}) -> found: {}",
+            tx_id,
+            tx.txid == *tx_id
+        );
+        Ok(tx)
+    }
+
+    fn get_raw_transaction_verbosity_two(
```

Comment:
You could consider renaming this method to `get_raw_full_transaction`.  
If the running Bitcoin Core version is 25.0, use it; otherwise, throw an error.

Reply:
i will stick to get_raw_transaction_verbosity_two as is more informative for the caller, which verbosity is using in the underlying rpc, and it is also aligned with alpen labs implementation

Using an old version of bitcoin core, will throw the RPC error, as all the other funcs when using a very old node, so the errors are consistent, therefore we wont add a new specific error only for this version